### PR TITLE
Fix the deep linking message to be compliant with the spec

### PR DESCRIPTION
**Pull request by marcospri**
Canvas seems to ignore this value so sending the wrong value hasn't had
any effect but the IMS certification tools does check the value and
other implementations might.

# Testing 

- Deeplinking continues working on LTI1.3 on Canvas.

### CODE REVIEW

File: tests/unit/lms/views/lti/deep_linking_test.py [hunk 1]
```diff
@@ -69,11 +69,37 @@ def test_it_for_v13(
...
+        _get_content_url_mock,
+        pyramid_request,
+    ):
+        del pyramid_request.parsed_params["deep_linking_settings"]["data"]
+
+        views.file_picker_to_form_fields_v13()
+
+        ltia_http_service.sign.assert_called_once_with(
+            {
+                "https://purl.imsglobal.org/spec/lti/claim/deployment_id": application_instance.deployment_id,
```
Comment (marcospri) [inline tests/unit/lms/views/lti/deep_linking_test.py:91]: I only care about `https://purl.imsglobal.org/spec/lti-dl/claim/data` not being here, not sure if it's clear enough.

File: lms/views/lti/deep_linking.py [hunk 2]
```diff
@@ -106,24 +106,27 @@ def file_picker_to_form_fields_v13(self):
             name="application_instance"
         ).get_current()
 
-        url = self._get_content_url(self.request)
+        message = {
```
Comment (marcospri) [inline lms/views/lti/deep_linking.py:109]: Same message, without `https://purl.imsglobal.org/spec/lti-dl/claim/data`

File: lms/views/lti/deep_linking.py [hunk 3]
```diff
@@ -106,24 +106,27 @@ def file_picker_to_form_fields_v13(self):
...
+
+        if data := self.request.parsed_params["deep_linking_settings"].get("data"):
+            # From: https://www.imsglobal.org/spec/lti-dl/v2p0#deep-linking-response-message
+            #
+            # The https://purl.imsglobal.org/spec/lti-dl/claim/data value must match the value
+            # of the data property of the https://purl.imsglobal.org/spec/lti-dl/claim/deep_linking_settings
+            # claim from the LtiDeepLinkinkingRequest message.
+            #
+            # This claim is required if present in LtiDeepLinkingRequest message.
+            message["https://purl.imsglobal.org/spec/lti-dl/claim/data"] = data
```
Comment (marcospri) [inline lms/views/lti/deep_linking.py:126]: Add it back if we have it from the request the platform sent us.

Comment (jon-betts) [inline lms/views/lti/deep_linking.py:126]:
```suggestion
        # From: 
        #  https://www.imsglobal.org/spec/lti-dl/v2p0#deep-linking-response-message
        #
        # The https://purl.imsglobal.org/spec/lti-dl/claim/data value must 
        # match the value of the data property of the 
        # https://purl.imsglobal.org/spec/lti-dl/claim/deep_linking_settings
        # claim from the `LtiDeepLinkinkingRequest` message.
        #
        # This claim is required if present in the `LtiDeepLinkingRequest` 
        # message.
        if data := self.request.parsed_params["deep_linking_settings"].get("data"):
            message["https://purl.imsglobal.org/spec/lti-dl/claim/data"] = data
```

Mostly trying to keep the code together here so I don't forget what the line above was by the time I've finished reading the comment. Tiny brain etc.

File: tests/unit/lms/views/lti/deep_linking_test.py [hunk 4]
```diff
@@ -69,11 +69,38 @@ def test_it_for_v13(
...
+                "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiDeepLinkingResponse",
+                "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+                "https://purl.imsglobal.org/spec/lti-dl/claim/content_items": [
+                    {
+                        "type": "ltiResourceLink",
+                        "url": _get_content_url_mock.return_value,
+                    }
+                ],
+            }
+        )
```
Comment (jon-betts) [inline tests/unit/lms/views/lti/deep_linking_test.py:102]:
```suggestion
    def test_it_for_v13_missing_deep_linking_settings_data(
        self, ltia_http_service, views, pyramid_request
    ):
        del pyramid_request.parsed_params["deep_linking_settings"]["data"]

        views.file_picker_to_form_fields_v13()

        ltia_http_service.sign.assert_called_once_with(message := Any.dict())
        assert (
            "https://purl.imsglobal.org/spec/lti-dl/claim/data "
            not in message.last_matched()
        )
```

Fancy new matcher stuff to the rescue.

**Review by jon-betts — APPROVED** [review]
I've made some suggestions for style, but the substance seems fine to me.

Not like we can really test it right now if Canvas doesn't care.

**Pull request merged**